### PR TITLE
New core

### PR DESCRIPTION
**Pull request by MeitarR**

### CODE REVIEW

File: run.py [hunk 1]
```diff
@@ -56,58 +72,55 @@ def generate_captcha():
...
-        await client.delete_message(message)
-    if message.author in unverified.keys():
-        if message.content == unverified[message.author]:
-            await client.send_message(message.channel,
-                                      message.author.mention + ' thanks!')
-            unverified.pop(message.author)
 
+@client.event
+async def on_message(message):
+    if message.author == client.user:
```
Comment (alumag) [inline run.py]: I think we should ignore any kind of bot.
`message.author.bot`

File: run.py [hunk 2]
```diff
@@ -56,58 +72,55 @@ def generate_captcha():
...
+        return
+    elif message.content.startswith(CMD_SIGN):
+        await process_cmd(message)
+    else:
```
Comment (alumag) [inline run.py]: we stop cursing people for saying "עד מתי"? :(

File: run.py [hunk 3]
```diff
@@ -1,50 +1,66 @@
...
 )
 
-from plugins import commands, async_commands, karma_store_cmds
-from plugins.karma import _take_karma
+
+MemberOnServer = namedtuple('MemberOnServer', 'user server')
+Command = namedtuple('Command', 'function channels')
 
 unverified = {}
+commands = {}
```
Comment (alumag) [inline run.py:24]: why it's located here?

File: plugins/rand_cmd.py [hunk 4]
```diff
@@ -1,17 +1,24 @@
...
-    if not (num1.isdigit() and num2.isdigit()):
-        return 'Illegal'
-    try:
-        return 'Result: ' + str(randint(float(num1), float(num2)))
-    except:
-        return 'LOL?'
+from cybot import utils, client
+
+
+@utils.register_command(name='rand', channels=['bot'])
```
Comment (alumag) [inline plugins/rand_cmd.py:8]: niceeeee

File: plugins/karma_store.py [hunk 5]
```diff
@@ -13,7 +13,12 @@
 ]
 
 
+@utils.register_command(name='buy', channels=['#karma-store'])
```
Comment (alumag) [inline plugins/karma_store.py:16]: what does the '#' mean? why sometimes we its just 'bot' and sometimes '#bot'?
I think it should be regex

**Review by alumag — CHANGES_REQUESTED** [review]

File: plugins/karma_store.py (hunk 5 above)
Comment (MeitarR) [reply plugins/karma_store.py:16]: .# means the exact name
without it means the string must be only part of the channel name

File: run.py (hunk 3 above)
Comment (MeitarR) [reply run.py:24]: for the process_cmd func

File: cybot/settings.py [hunk 6]
```diff
@@ -1,13 +1,13 @@
...
+
+DIR = os.path.dirname(__file__) + '/'
+CMD_SIGN = '$'
 
 user_kick_timeout = 700
 eng = "poiuytrewqlkjhgfdsamnbvcxz"
 heb = "פםןוטארק'/ךלחיעכגדשצמנהבסז"
-
-with open('help_file.txt') as help_file:
-    DOC_TXT = help_file.read()
```
Comment (alumag) [inline cybot/settings.py:13]: where do we load it? please explain the changes

File: cybot/utils.py [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+def admin(func):
+    @wraps(func)
+    def wrapped(message, args):
+        if message.author.server.owner.top_role not in message.author.roles:
+            return
+        return func(message, args)
+    return wrapped
+
+
+def is_right_channel(name_str, channel_name):
```
Comment (alumag) [inline cybot/utils.py]: should accept regex
i would also rename to "validate_channel"

File: run.py (hunk 2 above)
Comment (MeitarR) [reply run.py]: I just was lazy, we can add it again

File: run.py (hunk 1 above)
Comment (MeitarR) [reply run.py]: yea

File: cybot/utils.py [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+    """
+    functions_list = []
+
+    def registrar_warp(name, channels=None):
+        if channels is None:
+            the_channels = None
+        else:
+            the_channels = []
+            for ch_name in channels:
+                if ch_name.startswith('#'):
```
Comment (alumag) [inline cybot/utils.py:32]: wtf

**Review by alumag — CHANGES_REQUESTED** [review]

File: cybot/settings.py (hunk 6 above)
Comment (MeitarR) [reply cybot/settings.py:13]: the description of the command taken from the __doc__ of the function so the help file is unneeded

File: plugins/karma_store.py (hunk 5 above)
Comment (alumag) [reply plugins/karma_store.py:16]: in regex it would be just '^bot$' for the exact name or 'bot' for part of the channel name

File: run.py (hunk 3 above)
Comment (alumag) [reply run.py:24]: I think it's will be better to use `cybot` package for these kind of vars

Comment (MeitarR) [reply run.py:24]: but its var that used only on process_cmd and the loop at the outer scope that inserting all the commands from the register_command in the right format so process_cmd could use it

**Pull request merged**